pkg/kapp/diffgraph/change.go

```diff
@@ -232,3 +239,53 @@ func (cs Changes) MatchesRule(rule ChangeRule, exceptChange *Change) ([]*Change,
 
 	return result, nil
 }
+
+func (group *ChangeGroup) UpdateDefault(change *Change) {
+	if change.Change.Op() == ActualChangeOpUpsert {
+		switch group.Name {
+		case namespaceDefaultChangeGroup:
+			group.Name += "-" + change.Change.Resource().Name()
+		case crdDefaultChangeGroup:
+			group.Name += "-" + strings.ToLower(change.CRDKind())
+		}
+	}
+}
+
+func (rule *ChangeRule) UpdateDefault(change *Change) {
+	if change.Change.Op() == ActualChangeOpUpsert {
```

Comment:
Should we update delete ordering as well? This could lead to a bit of complexity as there are multiple change-rules associated with a single change-rule